docs/architecture/adr-001-poap-contract.md

```diff
@@ -0,0 +1,218 @@
+* The `base_poap_uri` identifies a valid `IPFS` URI corresponding to where the assets and metadata of the POAPs are stored.
+  * The `metadata` file is a `.json` that follow the `ERC-721` [metadata standard](https://docs.opensea.io/docs/metadata-standards#metadata-structure)
+* The field will be used to initialize the `token_uri` field of the `cw721-base` [`MintMsg<T>`](https://github.com/CosmWasm/cw-nfts/blob/1e992ccf640f07a384d6442625d6780a8e48ef1e/contracts/cw721-base/src/msg.rs#L61)
+* The `event_uri` field is used to initialize the `extension` part of the `cw721-base` [MintMsg< T >](https://github.com/CosmWasm/cw-nfts/blob/1e992ccf640f07a384d6442625d6780a8e48ef1e/contracts/cw721-base/src/msg.rs#L61) and contains a valid `IPFS` URI corresponding to where a `.json` file with the following event's metadata is stored:
+
+```json
+{
+  "name": "My awesome event",
+  "description": "Brief description of the event",
+  "city": "city where the event will be taking place",
+  "country": "country where the event will be taking place",
+  "start_date": "Start date for your event",
+  "end_date": "End date for your event",
+  "expiry_date": "From this day onward, no attendees will be able to mint POAPs from your event",
+  "year": <number>,
+  "event_url": "Event URL",
+  "virtual_event": true or false,
+}
+```
+
+#### Execute
+```rust
+pub enum ExecuteMsg {
+  EnableMint{},
+  DisableMint{},
+  Mint{},
+  MintTo{recipient: String},
+  UpdateEventInfo {
+    start_time: Timestamp,
+    end_time: Timestamp,
+  },
+  UpdateAdmin{new_admin: String},
+  UpdateMinter{new_minter: String}
+}
+```
+
+#### EnableMint
+With the `EnableMint{}` message the admin can enable the `Mint{}` message for everyone.
+
+#### DisableMint
+With the `DisableMint{}` message the admin can disable the `Mint{}` message for everyone.
+
+#### Mint
+If enabled, the `Mint{}` message allows users to mint their own POAP.
+
+#### MintTo
+With the `MintTo{recipient}` message the contract's admin or the minter can mint the POAP for a specific recipient.
+
+#### UpdateEventInfo
+With the `UpdateEventInfo{start_time, end_time}` message the event's creator can change the time frame of the event if it's not already started or finished.
+
+#### UpdateAdmin
+With the `UpdateAdmin{new_admin}` message the contract's admin can transfer the admin rights to another user.
+
+#### UpdateMinter
+With the `UpdateMinter{new_minter}` message the contract's admin can choose another minter to which give the minting
+rights.
+
+### Query
+All the queries below, except for the `EventInfo`, `Admin` and `MintStatus` are inherited from [cw721-base queries](https://github.com/CosmWasm/cw-nfts/blob/1e992ccf640f07a384d6442625d6780a8e48ef1e/contracts/cw721-base/src/msg.rs#L76).
+```rust
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+
+    /// Inherited from CW721-base
```

Comment:
All the CW721-base queries can be reacted by the CW721-base target contract address. We can show the target CW721 contract address, instead.